### PR TITLE
Cart & Checkout

### PR DESCRIPTION
**Pull request by katemorris**
- Adds incrementality to the cart for items. 
- Adds test to checkout for the full checkout. 
- Changes the create order action to do everything correctly with users attached. 
- **There are broken tests in this PR that will be fixed with Shaunda's PR**
- This does not have exactly the same setup as Shaunda, so I expect her PR to overwrite some routes, the order profile index page location and one method in that page I didn't want to ask for (I asked her for a lot, sorry S).

Two main test files were:
- spec/features/cart/show_spec.rb
- spec/features/cart/checkout_spec.rb

Closes US 23-26

### CODE REVIEW

File: config/routes.rb
Comment (jdcarey128) [inline config/routes.rb:40]: Is it necessary to break from RESTful convention here?

**Review by jdcarey128 — CHANGES_REQUESTED** [review]

**Comment by jdcarey128** [comment]
Can you resolve the merge conflict locally before we merge? I can walk you through that at 5 if you want.

File: config/routes.rb
Comment (katemorris) [reply config/routes.rb:40]: I was figuring that the above already broke restful (add_item is not restful), and restful has a patch and post verb to the same path, I just changed the action. Open to suggestions!

**Comment by katemorris** [comment]
Yep, planning on resolving the merge conflict when we are done chatting!

**Comment by katemorris** [comment]
I think I got it @jdcarey128 - please check it out. I kept your change for password and added the relationship to orders.

**Review by jdcarey128 — APPROVED** [review]

**Review by smcunning — APPROVED** [review]

File: config/routes.rb
Comment (taphill) [reply config/routes.rb:40]: I agree that they didn't really set us up for successfully using RESTful routing for the cart controller which is odd. I think this could be refactored **if** we wanted to go down that path.

In my mind we'd have to do something like Cart::ItemsController and then use the standard CRUD actions from there...so `add_item` would be `new`, `remove_item` would be `destroy` and `change_quantity` would be `update`

**Review by taphill — APPROVED** [review]
Overall this looks fine to me! I did leave some comments about the RESTful routing thing...not sure what the thoughts were on that, or if you all just wanted to move on?

File: config/routes.rb
Comment (katemorris) [reply config/routes.rb:40]: I say we leave it to a refactor. I'll put this in a card and you can merge this.

Comment (katemorris) [reply config/routes.rb:40]: Updated!

**Pull request merged**
